sw/host/opentitanlib/src/util/file.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use nix::libc::c_int;
+use nix::poll;
+use std::cmp;
+use std::io;
+use std::os::unix::io::AsRawFd;
+use std::time::Duration;
+
+/// Waits for an event on `fd` or for `timeout` to expire.
+pub fn wait_timeout(fd: &impl AsRawFd, events: poll::PollFlags, timeout: Duration) -> Result<()> {
+    let timeout = cmp::min(timeout.as_millis(), c_int::max_value() as u128) as c_int;
```

Comment:
Is this clearer (type annotation *probably* unnecessary)?

```rust
use std::convert::TryInto;
let timeout: c_int = timeout.as_millis().try_into().unwrap_or(c_int::MAX);
```

Reply:
Yes, I think that looks a bit better.